pkg/cmd/pulumi/state/state_unprotect.go

```diff
@@ -106,6 +109,70 @@ func unprotectAllResources(ctx context.Context, ws pkgWorkspace.Context, stackNa
 	return nil
 }
 
+// unprotectMultipleResources unprotects multiple resources specified by their URNs.
+func unprotectMultipleResources(
+	ctx context.Context, ws pkgWorkspace.Context, stackName string, urns []string, showPrompt bool,
+) error {
+	opts := display.Options{
+		Color: cmdutil.GetGlobalColorization(),
+	}
+	s, err := cmdStack.RequireStack(
+		ctx,
+		ws,
+		backend.DefaultLoginManager,
+		stackName,
+		cmdStack.OfferNew,
+		opts,
+	)
+	if err != nil {
+		return err
+	}
+
+	return TotalStateEdit(ctx, s, showPrompt, opts, func(_ display.Options, snap *deploy.Snapshot) error {
+		if snap == nil {
+			return errors.New("no resources found to unprotect")
+		}
+
+		var errs []string
+		resourceCount := 0
+
+		for _, urnStr := range urns {
+			urn := resource.URN(urnStr)
+			found := false
+
+			for _, res := range snap.Resources {
+				if res.URN == urn {
+					found = true
+					if err := edit.UnprotectResource(snap, res); err != nil {
+						errs = append(errs, fmt.Sprintf("error unprotecting resource %q: %v", urn, err))
+					} else {
+						resourceCount++
+					}
+					break
+				}
+			}
+
+			if !found {
+				errs = append(errs, fmt.Sprintf("No such resource %q exists in the current state", urn))
+			}
+		}
+
+		if resourceCount > 0 {
+			if len(errs) > 0 {
+				fmt.Printf("%d resources unprotected with errors\n", resourceCount)
```

Comment:
This message isn't correct.  If we return an error in the callback we pass to `TotalStateEdit`, we don't write the state at all, so no resources will be unprotected (see https://github.com/pulumi/pulumi/blob/master/pkg/cmd/pulumi/state/io.go#L171)

Reply:
Bump, I still don't think this is correct.